docs/guide/request_lifecycle.md

```diff
@@ -0,0 +1,225 @@
+The `ContentViewBuilder` (`ezpublish.view_builder.content`) matches on controller starting with `ez_content:` (see `eZ\Publish\Core\MVC\Symfony\View\Builder\ContentViewBuilder::matches`).
+The `ContentViewBuilder` builds a `ContentView`.
+
+First, the `ContentViewBuilder` will load the `Location` and the `Content`, and, add them to the `ContentView` object.
+
+**Notice about Permission Control**: `content/read` and/or `content/view_embed` permissions are controlled during this `ContentView` building.
+
+Then, the `ContentViewBuilder` passes the `ContentView` to its `View\Configurator` (`ezpublish.view.configurator`).
+It's implemented by the `View\Configurator\ViewProvider` and its `View\Provider\Registry`, this registry receives the services tagged `ezpublish.view_provider` thanks to the `ViewProviderPass`.
+Among the view providers, the services using the `eZ\Bundle\EzPublishCoreBundle\View\Provider\Configured` have an implementation of the `MatcherFactoryInterface` (`ezpublish.content_view.matcher_factory`).
+Through service decoration and class inheritance, the `ClassNameMatcherFactory` will be responsible for the [View Matching](content_rendering/templates/template_configuration/#view-rules-and-matching).
+The `View\Configurator\ViewProvider` will use the matched view rule to add possible **`templateIdentifier`** and **`controllerReference`** to the `ContentView` object.
+
+The `ViewControllerListener` adds the ContentView to the `Request` as **`view`** attribute.
+The `ViewControllerListener` eventually updates the `Request` `_controller` attribute with the `ContentView`'s `controllerReference`.
+
+The `HttpKernel` then dispatches a `kernel.controller_arguments` (a.k.a. `KernelEvents::CONTROLLER_ARGUMENTS`) but nothing from Ibexa DXP is listening to it.
+
+
+
+## Controller execution
+
+The `HttpKernel` extracts from the Request the controller and the arguments to pass to the controller. [Argument resolvers](https://symfony.com/doc/current/controller/argument_value_resolver.html) way is similar to autowiring.
+The `HttpKernel` executes the controller with those arguments.
+
+As a reminder, the controller and its argument can be:
+
+- A controller set by a route matching and the `Request` as its argument.
+- The default `ez_content:viewAction` controller and a `ContentView` as its argument.
+- A [custom controller](content_rendering/queries_and_controllers/controllers/) set by the matched view rule and a `View` or the `Request` as its argument (most likely a `ContentView` but there is no restriction). **Notice about Permission Control**: See [Permissions for custom controller](https://doc.ibexa.co/en/latest/guide/permissions/#permissions-for-custom-controllers).
```

Comment:
```suggestion
- A [custom controller](content_rendering/queries_and_controllers/controllers.md) set by the matched view rule and a `View` or the request as its argument (most likely a `ContentView` but there is no restriction).

!!! caution "Permission Control"

    See [Permissions for custom controller](permissions.md#permissions-for-custom-controllers).
```
Hint: internal links should ideally be indicated used with `.md` instead of just `/`, as otherwise the automatic link checker does not work.